### DISCUSSION TITLE
Question regarding edit the email and description on the swagger ui

### DISCUSSION THREAD

**Discussion by callmedada** [bentoml/BentoML#2367] [Q&A] [answered]
Hi,
First of all, thanks a lot for developing this powerful tool. I am currently trying to edit the swagger ui to add the developer email, description and the readme. But I could not able to find a wiki regarding that and wondering how to edit it. 
I.e. to edit  info = {
        "title": svc.name,
        "description": "A Prediction Service built with BentoML",
        "contact": {"email": "contact@bentoml.ai"},
        "version": svc.version or "0.0.0",
    }
at https://github.com/bentoml/BentoML/blob/main/bentoml/_internal/service/openapi.py#L48

Any comments would be much appreciated. 

Thanks so much,
Dennis

**Comment by aarnphm** [comment, marked as answer]
Hi @callmedada, You can customize openapi docs saved under the bento directory at `openapi.yml`.

**Reply by callmedada** [reply]
Thanks so much for the answer! Exactly what I am looking for.

**Reply by aarnphm** [reply]
happy to help. Let me know if you have any other question

**Reply by callmedada** [reply]
Hey @aarnphm ,

I am so sorry but just a follow up questions. When I containerize the BentOS and run it on openshift, seems like it will just ignore the openapi.yaml (I have all definitions in openapi.yaml but the UI just does not show). I used the based image: bento-server:1.0.0a6-python3.9-debian-runtime.

Thanks again

**Comment by parano** [comment]
@aarnphm @callmedada actually changing the `openapi.yaml` file in a Bento directory does not work and we do not recommend users to do that.

If all you need to do is add some additional documentation to the swagger page, what you could do is use the `description` field in the `bentofile.yaml` definition, it can be pointing to a local markdown file as well. the description will show up in the index page of your bento server.

We will consider adding a more flexible API for users to customize the generated open API doc, however it is not currently supported.

Another approach you could try, is to use the advanced API `mount_wsgi_app` and `mount_asgi_app` to customize your BentoServer to use a own index page. You can use a web framework like FastAPI or Flask to build it and it will be served together with other BentoServer endpoints. This way you can define any landing page you need, and use your own open openapi docs: https://github.com/bentoml/BentoML/tree/main/bentoml/_internal/server#mounting-a-flask-app-to-service

**Reply by callmedada** [reply]
Hi @parano ,

Thanks so much for your reply. Its good to know that we have the mount_wsgi_api. I guess BentoML has many many great stuffs we do not realize. Thanks so much again,
Dennis
